File: tests/testdata/codegen/components-pp/python/simpleComponent.py
import khulnasoft
from khulnasoft import Input
from typing import Optional, Dict, TypedDict, Any
import khulnasoft_random as random

class SimpleComponent(khulnasoft.ComponentResource):
    def __init__(self, name: str, opts: Optional[khulnasoft.ResourceOptions] = None):
        super().__init__("components:index:SimpleComponent", name, {}, opts)

        first_password = random.RandomPassword(f"{name}-firstPassword",
            length=16,
            special=True,
            opts = khulnasoft.ResourceOptions(parent=self))

        second_password = random.RandomPassword(f"{name}-secondPassword",
            length=16,
            special=True,
            opts = khulnasoft.ResourceOptions(parent=self))

        self.register_outputs()

File: tests/testdata/codegen/components-pp/nodejs/simpleComponent.ts
import * as khulnasoft from "@khulnasoft/khulnasoft";
import * as random from "@khulnasoft/random";

export class SimpleComponent extends khulnasoft.ComponentResource {
    constructor(name: string, opts?: khulnasoft.ComponentResourceOptions) {
        super("components:index:SimpleComponent", name, {}, opts);
        const firstPassword = new random.RandomPassword(`${name}-firstPassword`, {
            length: 16,
            special: true,
        }, {
            parent: this,
        });

        const secondPassword = new random.RandomPassword(`${name}-secondPassword`, {
            length: 16,
            special: true,
        }, {
            parent: this,
        });

        this.registerOutputs();
    }
}
